src/map-events.tsx

```diff
@@ -105,17 +106,14 @@ export type Listeners = {
 
 export const listenEvents = (
   partialEvents: EventMapping,
-  props: Partial<Events>,
+  parent: React.Component,
```

Comment:
We can't pass in `props` because then the closure traps an out-of-date value when the component re-renders. It needs to always grab the latest value off of the `this` object.